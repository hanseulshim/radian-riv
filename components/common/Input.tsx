import React, { useState } from 'react'

interface Props {
  value: string | number
  label?: string
  type?: string
  required?: boolean
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  error: string
  disabled?: boolean
}

export default function Input({
  value,
  label,
  type = 'text',
  required = false,
  onChange,
  error,
  disabled = false
}: Props) {
  const [focus, setFocus] = useState(false)
  return (
    <div className="input-group">
      <input
        className={error ? 'error-input' : null}
        type={type}
        placeholder={label && `${required ? '   ' : ''}${label}...`}
        name={label}
        value={value}
        onChange={onChange}
        onFocus={() => setFocus(true)}
        onBlur={() => setFocus(false)}
        disabled={disabled}
      />
      {required && !focus && !value && (
        <span className="placeholder required-field">*</span>
      )}
      {label && (
        <label className={value ? 'input-label' : null} htmlFor={label}>
          {required && <span className="required-field">* </span>}
          {label}
        </label>
      )}
      <span className="error-input-message">{error}</span>
    </div>
  )
}
