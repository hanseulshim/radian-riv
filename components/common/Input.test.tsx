import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Input from './Input'

describe('Input', () => {
  let component
  beforeAll(() => {
    const props = {
      value: 'test',
      label: 'test',
      onChange: () => {},
      error: 'error'
    }
    component = shallow(<Input {...props} />)
  })
  it('renders input-group', () => {
    expect(component.find('.input-group')).toHaveLength(1)
  })
  it('renders error-input', () => {
    expect(component.find('input.error-input')).toHaveLength(1)
  })
  it('renders label', () => {
    expect(component.find('label.input-label').text()).toEqual('test')
  })
  it('renders error message', () => {
    expect(component.find('span.error-input-message').text()).toEqual('error')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
