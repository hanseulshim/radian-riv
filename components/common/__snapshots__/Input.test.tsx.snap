// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<div
  className="input-group"
>
  <input
    className="error-input"
    disabled={false}
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="test..."
    type="text"
    value="test"
  />
  <label
    className="input-label"
    htmlFor="test"
  >
    test
  </label>
  <span
    className="error-input-message"
  >
    error
  </span>
</div>
`;
